PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/PluginUtil.java

```diff
@@ -430,4 +452,146 @@ public static Image getImage(String imagePath) {
 			return null;
 		}
 	}
+	
+	public static void forceInstallPluginUsingMarketPlaceAsync(String pluginSymbolicName, String marketplaceURL) {
+		Display.getDefault().asyncExec(new Runnable() {
+
+			@Override
+			public void run() {
+				forceInstallPluginUsingMarketplace(pluginSymbolicName, marketplaceURL);
+			}
+			
+		});;
+	}
+	/**
+	 * This function checks whether an Eclipse plug-in is installed or not. 
+	 * If not, it will firstly check whether Eclipse Marketplace Client plug-in is installed and install it
+	 * Then it will use Marketplace Client to install the plug-in
+	 * 
+	 * @param pluginSymbolicName: Symbolic name for the plug-in, for example, org.eclipse.fx.ide.feature for e(fx)clipse, org.scala-ide.sdt.core for Scala
+	 * @param marketplaceURL: Marketplace installation URL, for example, http://marketplace.eclipse.org/marketplace-client-intro?mpc_install=421 for Scala
+	 * @return
+	 */
+	public static boolean forceInstallPluginUsingMarketplace(String pluginSymbolicName, String marketplaceURL) {
+		boolean isTargetInstalled = checkPlugInInstallation(pluginSymbolicName);
+		if (isTargetInstalled == true) {
+			return true;
+		}
+			
+		boolean isMarketplacePluginInstalled = checkPlugInInstallation(marketplacePluginSymbolicName);
+		if (isMarketplacePluginInstalled != true) {
+			PluginUtil.displayErrorDialog(getParentShell(), "Install missing plugin", "Start to install Eclipse Marketplace Client plugin which is required to install other missing plugin (" + pluginSymbolicName + ")!");
+			forceInstallPluginUsingP2(marketplacePluginID);		
+		}
+	
+		try {
+			PluginUtil.displayErrorDialog(getParentShell(), "Install missing plugin", "Start to install missing plugin (" + pluginSymbolicName + ")!");
+			SolutionInstallationInfo info = MarketplaceUrlHandler.createSolutionInstallInfo(marketplaceURL);
+			MarketplaceUrlHandler.triggerInstall(info);
+		} catch (Exception e) {
+			String errorMsg = "Error installing " + pluginSymbolicName + "! Please manually install using Eclipse marketplace from: Help -> Eclipse Marketplace....";
+			PluginUtil.displayErrorDialogAndLog(getParentShell(), "Fail to install", errorMsg, e);
+			
+			try {
+				MarketplaceClient.openMarketplaceWizard(null);
```

Comment:
This will open the marketplace UI and let users to search the plugin to install

---
In reply to: [132093696](https://github.com/Microsoft/azure-tools-for-java/pull/845#discussion_r132093696) [](ancestors = 132093696)